frontend/src/components/Main.js

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import NewPageModal from "./NewPageModal";
+import PageContent from "./PageContent";
+import PageTree from "./PageTree";
+import WikiList from "./WikiList";
+
+class Main extends React.Component {
```

Comment:
React [doesn't recommend rewriting class components][0] but [recommends you write new components in the new "hooks" style][1]. (Hooks are a conceptual whirlwind, as you've probably seen.)

[0]: https://reactjs.org/docs/hooks-faq.html#do-i-need-to-rewrite-all-my-class-components
[1]: https://reactjs.org/docs/hooks-faq.html#should-i-use-hooks-classes-or-a-mix-of-both

Reply:
BAH! I thought I was reading that this _WAS_ the new way to do this 🤦 I swear I saw "legacy hooks" pattern somewhere. I was writing hooks for a while until it got to that.